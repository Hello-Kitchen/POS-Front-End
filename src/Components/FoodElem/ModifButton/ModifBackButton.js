import { GoArrowDown } from "react-icons/go";
import { useNavigate } from "react-router-dom";

function ModifButton({id, food}) {

    const navigate = useNavigate();
    const handleClick = () => {
        navigate(-1, {state: {food: food}})
    }

    return (
        <div className="h-1/6 w-full grid grid-flow-col colbottom-0 p-6">
            <div className="w-full col-span-5 justify-items-center flex-row content-center">
                <h1 className="text-3xl font-bold text-black">
                    Modifications
                </h1>
            </div>
            <div className="col-span-1 w-full grid justify-items-end float-right">
                <button onClick={() => handleClick()}>
                    <GoArrowDown size={40} color="black" />
                </button>
            </div>
        </div>
    )
}

export default ModifButton;