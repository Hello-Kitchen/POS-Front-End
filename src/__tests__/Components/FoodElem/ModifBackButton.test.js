import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import ModifBackButton from '../../../Components/FoodElem/ModifButton/ModifBackButton';
import { useNavigate } from 'react-router-dom';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(),
}));

describe('ModifBackButton Component', () => {
  
    test('renders', () => {
        const { getByText, getByRole } = render(
            <Router>
                <ModifBackButton />
            </Router>
        );

        expect(getByText('Modifications')).toBeInTheDocument();

        const button = getByRole('button');
        expect(button).toBeInTheDocument();

        const icon = button.querySelector('svg');
        expect(icon).toBeInTheDocument();
    });

    test('navigate', () => {
        const mockNavigate = jest.fn();
        useNavigate.mockReturnValue(mockNavigate);

        const { getByRole } = render(
            <Router>
                <ModifBackButton />
            </Router>
        );

        const button = getByRole('button');
        fireEvent.click(button);

        expect(mockNavigate).toHaveBeenCalledWith(-1);
    });

    test('matches the snapshot', () => {
        const { asFragment } = render(
            <Router>
                <ModifBackButton />
            </Router>
        );

        expect(asFragment()).toMatchSnapshot();
    });
}) ;
