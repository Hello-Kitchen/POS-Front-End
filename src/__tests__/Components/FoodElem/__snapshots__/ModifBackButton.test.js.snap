// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModifBackButton Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full row-span-1 grid grid-flow-col grid-cols-6 colbottom-0 content-center pl-6 pr-6"
  >
    <div
      class="w-full col-span-5 justify-items-center flex-row content-center"
    >
      <h1
        class="text-3xl font-bold text-black"
      >
        Modifications
      </h1>
    </div>
    <div
      class="col-span-1 w-full grid justify-items-end float-right"
    >
      <button>
        <svg
          color="black"
          fill="currentColor"
          height="40"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.97 13.22a.75.75 0 0 1 1.06 0L11 18.19V3.75a.75.75 0 0 1 1.5 0v14.44l4.97-4.97a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734l-6.25 6.25a.75.75 0 0 1-1.06 0l-6.25-6.25a.75.75 0 0 1 0-1.06Z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
